### ISSUE TITLE
Unable to generate a client that uses multi-level `deepObject` serialization 

### ISSUE THREAD

**Issue by greglanthier** [greglanthier/gocflapi#2]
# TL;DR:

Coming up with a way to encode a multi-dimensional deep object query parameter in an OpenAPI specification is not well defined right now (see https://github.com/OAI/OpenAPI-Specification/issues/1706)

This will make it difficult to generate a 100% complete API client from an OpenAPI spec that fully supports the [CFL API](http://api.cfl.ca/docs).

## Background

The [CFL API](http://api.cfl.ca/docs) operations that use [filters](http://api.cfl.ca/docs#games-filters) rely on what the OpenAPI specification calls [deepObject serialization](https://swagger.io/docs/specification/serialization/#query).  The current OpenAPI specification only covers the case where the object in question is one level deep, i.e.: 

`.../foo?filter[key]=value`

```json
{"key":"value"}
```

The [filters](http://api.cfl.ca/docs#games-filters) on the Games APIs are two levels deep, i.e.:

`/v1/games?filter[team][eq]=TOR`

```json
{
  "team" : {
    "eq" : "TOR"
  }
}
```

Until I come up with a workaround for this it will not be possible to support filter operations in the [CFL API](http://api.cfl.ca/docs).
